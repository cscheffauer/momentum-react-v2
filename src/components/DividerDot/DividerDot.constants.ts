const CLASS_PREFIX = 'md-divider-dot';

const DEFAULTS = {
  SIZE: 'small' as const,
};

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, DEFAULTS, STYLE };
