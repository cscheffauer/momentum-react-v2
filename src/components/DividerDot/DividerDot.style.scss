.md-divider-dot-wrapper {
  display: inline-block;
}
