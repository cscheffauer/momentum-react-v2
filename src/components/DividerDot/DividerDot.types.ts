import { CSSProperties } from 'react';

export interface Props {
  /**
   * Size of the Divider dot
   *
   * @default 'small'
   */
  size?: 'small' | 'medium' | 'large';

  /**
   * Custom class for overriding this component's CSS.
   */
  className?: string;

  /**
   * Custom id for overriding this component's CSS.
   */
  id?: string;

  /**
   * Custom style for overriding this component's CSS.
   */
  style?: CSSProperties;
}
