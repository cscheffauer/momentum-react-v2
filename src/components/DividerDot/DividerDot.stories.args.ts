import { commonStyles } from '../../storybook/helper.stories.argtypes';
import { DEFAULTS } from './DividerDot.constants';

export default {
  ...commonStyles,
  size: {
    description: 'To set the size of DividerDot',
    options: ['small', 'medium', 'large'],
    control: { type: 'select' },
    table: {
      type: {
        summary: 'string',
      },
      defaultValue: {
        summary: DEFAULTS.SIZE,
      },
    },
  },
};
