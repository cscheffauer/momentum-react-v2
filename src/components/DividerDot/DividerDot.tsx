import React, { FC } from 'react';
import classnames from 'classnames';

import { Bullet as MdcBullet } from '@momentum-design/components/dist/react';
import type { Props } from './DividerDot.types';
import { DEFAULTS, STYLE } from './DividerDot.constants';
import './DividerDot.style.scss';

/**
 * The DividerDot component.
 */
const DividerDot: FC<Props> = (props: Props) => {
  const { className, id, style, size = DEFAULTS.SIZE } = props;

  return (
    <MdcBullet size={size} className={classnames(className, STYLE.wrapper)} id={id} style={style} />
  );
};

export default DividerDot;
