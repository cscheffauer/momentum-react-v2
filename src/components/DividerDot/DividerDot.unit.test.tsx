import React from 'react';
import { render, waitFor } from '@testing-library/react';

import DividerDot, { DIVIDER_DOT_CONSTANTS as CONSTANTS } from './';

describe('<DividerDot />', () => {
  const setup = async (component: any) => {
    const { container } = render(component);

    // we have to wait for the web component to be rendered
    await waitFor(() => {
      expect(container.querySelector('mdc-bullet')).toBeTruthy();
    });

    return container;
  };

  describe('snapshot', () => {
    it('should match snapshot', async () => {
      expect.assertions(2);

      const container = await setup(<DividerDot />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', async () => {
      expect.assertions(2);

      const className = 'example-class';

      const container = await setup(<DividerDot className={className} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', async () => {
      expect.assertions(2);

      const id = 'example-id';

      const container = await setup(<DividerDot id={id} />);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', async () => {
      expect.assertions(2);

      const style = { color: 'pink' };

      const container = await setup(<DividerDot style={style} />);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      expect.assertions(2);

      const container = await setup(<DividerDot />);
      const element = container.querySelector('mdc-bullet');

      expect(element.classList.contains(CONSTANTS.STYLE.wrapper)).toBe(true);
    });

    it('should have provided class when className is provided', async () => {
      expect.assertions(2);

      const className = 'example-class';

      const container = await setup(<DividerDot className={className} />);
      const element = container.querySelector('mdc-bullet');

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', async () => {
      expect.assertions(2);

      const id = 'example-id';

      const container = await setup(<DividerDot id={id} />);
      const element = container.querySelector('mdc-bullet');

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', async () => {
      expect.assertions(2);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      const container = await setup(<DividerDot style={style} />);
      const element = container.querySelector('mdc-bullet');

      expect(element.getAttribute('style')).toBe(styleString);
    });

    it('should have provided size when size is provided', async () => {
      expect.assertions(2);

      const size = 'large';

      const container = await setup(<DividerDot size={size} />);
      const element = container.querySelector('mdc-bullet');

      expect(element.getAttribute('size')).toBe(size);
    });
  });
});
