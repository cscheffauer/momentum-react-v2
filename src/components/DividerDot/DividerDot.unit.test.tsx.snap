// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DividerDot /> snapshot should match snapshot 1`] = `
<div>
  <mdc-bullet
    class="md-divider-dot-wrapper"
    size="small"
  />
</div>
`;

exports[`<DividerDot /> snapshot should match snapshot with className 1`] = `
<div>
  <mdc-bullet
    class="example-class md-divider-dot-wrapper"
    size="small"
  />
</div>
`;

exports[`<DividerDot /> snapshot should match snapshot with id 1`] = `
<div>
  <mdc-bullet
    class="md-divider-dot-wrapper"
    id="example-id"
    size="small"
  />
</div>
`;

exports[`<DividerDot /> snapshot should match snapshot with style 1`] = `
<div>
  <mdc-bullet
    class="md-divider-dot-wrapper"
    size="small"
    style="color: pink;"
  />
</div>
`;
